File: 240612/연속부분수열일까/contiguous-array-or-not.py
n1, n2 = map(int, input().split())
a_lst = list(map(int, input().split()))
b_lst = list(map(int, input().split()))

if n1 < n2:
    a_lst, b_lst = b_lst, a_lst

check = False
for i in range(n1 - n2 + 1):
    if a_lst[i: i + n2] == b_lst:
        check = True
        break

print("Yes" if check == True else "No")

File: 251118/연속부분수열일까/contiguous-array-or-not.cpp
#include <bits/stdc++.h>
using namespace std;

int main() 
{
    int n1, n2;
    cin >> n1 >> n2;

    vector<int> a_lst(n1), b_lst(n2);
    for (int i = 0; i < n1; i++) cin >> a_lst[i];
    for (int i = 0; i < n2; i++) cin >> b_lst[i];

    if (n1 < n2) 
    {
        swap(a_lst, b_lst);
        swap(n1, n2);
    }

    bool check = false;

    for (int i = 0; i <= n1 - n2; i++) 
    {
        bool same = true;
        for (int j = 0; j < n2; j++) 
        {
            if (a_lst[i + j] != b_lst[j]) 
            {
                same = false;
                break;
            }
        }
        if (same) 
        {
            check = true;
            break;
        }
    }

    cout << (check ? "Yes" : "No") << "\n";
    return 0;
}
